bibernate-core/src/main/java/com/bobocode/petros/bibernate/session/CacheableSession.java

```diff
@@ -24,28 +26,28 @@ public <T> T persist(T entity) {
     }
 
     @Override
-    public <T> T findById(Class<T> type, Object id) {
+    public <T> Optional<T> findById(Class<T> type, Object id) {
         return super.findById(type, id);
     }
 
     @Override
-    public <T> T find(Class<T> type, String propertyName, Object value) {
+    public <T> Collection<T> find(Class<T> type, String propertyName, Object value) {
         return super.find(type, propertyName, value);
     }
 
     @Override
     public <T> T update(T entity) {
-        return super.update(entity);
+        return jdbcQueryManager.update(entity);
```

Comment:
why do we call jdbcQueryManager here? It supposed to be a call to super.update(), isn't it? The same applies for a couple of methods below.